code/measures/nml_parseutil.py

```diff
@@ -0,0 +1,132 @@
+import datetime
+import xml.etree.ElementTree as ET
+from dateutil import parser
+import sys
+
+class Article:
+    
+    def __init__(self, company, timestamp, headline, article_text):
+        self.company = company
+        self.timestamp = parser.parse(timestamp)
+        self.headline = headline
+        self.article_text = article_text
+    
+    def __repr__(self):
+        return self.company + "; " + str(self.timestamp) + "; " + self.headline
+    
+    def __eq__(self, other):
+        if isinstance(other, Article):
+            return (self.company == other.company and self.timestamp == other.timestamp and
+                self.headline == other.headline)
+        return False
+    
+    def __lt__(self, other):
+        """
+        Comparison operator for sorting Articles by timestamp. 
+        """
+        return self.timestamp < other.timestamp
+    
+    def __hash__(self):
+        return hash(self.__repr__())
+    
+    def elapsed_time_between(self, other):
+        """
+        Returns the amount of time, in seconds, between the publishing 
+        of this Article and another Article, other. 
+        """
+        elapsed_time = self.timestamp - other.timestamp
+        return abs(elapsed_time.total_seconds())
+
+def filter_old_articles(company_article_map, curr_article, k_hours=72):
+    """
+    Check the set of articles mapped to curr_article.company, and 
+    filter out articles that are at least k_hours older than curr_article. 
+    
+    Arguments:
+        curr_article: An Article object, used to compare time stamps with all other
+            articles in company_article_map[curr_article.company]. 
+        k_hours: An int, default 72. All articles in company_article_map[curr_article.company] 
+            that have at least k_hours elapsed time between that article and 
+            curr_article.timestamp will be filtered from the map.
+            
+    Returns:
+        company_article_set: A new set to map to curr_article.company, with 
+            old articles filtered out, with "old" specified as k_hours.
+    """
+    
+    k_seconds = k_hours * 60 * 60
+    curr_company = curr_article.company
+    curr_timestamp = curr_article.timestamp
+    
+    company_article_set = company_article_map.get(curr_company, "No articles for this company")
+    
+    if company_article_set == "No articles for this company":
+        return set([curr_article])
```

Comment:
Return empty set to trigger logic: if empty, move to cleaning and updating company-story mapping, else, run sim_measure first.